### PR TITLE
Support polmorphic as

### PR DESCRIPTION
**Pull request by yakjuly**
has_many ***, as: ***
has_many :through

Support polymorphic assoication

### CODE REVIEW

File: spec/support/data.rb [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+post2 = Post.create(title: "Second post!")
+post3 = Post.create(title: "Third post!")
+
+post1.comments.create(status: 'created', rating: 4, author: student1)
+post1.comments.create(status: 'approved', rating: 5, author: student1)
+post1.comments.create(status: 'deleted', rating: 0, author: student2)
+
+post2.comments.create(status: 'approved', rating: 3, author: student1)
+post2.comments.create(status: 'approved', rating: 5)
+
```
Comment (flyerhzm) [inline spec/support/data.rb]: author is polymorphic, you should set it as different models like student and teacher to make it more sense.

File: spec/support/data.rb [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+teacher3 = Teacher.create name: "Teacher 3"
 student1 = Student.create name: 'Student 1'
 student2 = Student.create name: 'Student 2'
 student3 = Student.create name: 'Student 3'
 student4 = Student.create name: 'Student 4'
 teacher1.students = [student1]
 teacher2.students = [student2, student3, student4]
+
+
+
```
Comment (flyerhzm) [inline spec/support/data.rb]: one line break is enough

File: lib/eager_group/preloader.rb [hunk 3]
```diff
@@ -34,5 +44,6 @@ def run
         end
       end
     end
+
```
Comment (flyerhzm) [inline lib/eager_group/preloader.rb]: no extra line break here

File: spec/support/schema.rb [hunk 4]
```diff
@@ -12,6 +12,9 @@
   create_table :comments, :force => true do |t|
     t.string :body
     t.string :status
+
```
Comment (flyerhzm) [inline spec/support/schema.rb]: no extra line break

File: spec/support/models.rb [hunk 5]
```diff
@@ -21,6 +22,12 @@ class Teacher < ActiveRecord::Base
 class Student < ActiveRecord::Base
   has_many :classrooms
   has_many :teachers, through: :classrooms
+
```
Comment (flyerhzm) [inline spec/support/models.rb]: no extra line break

File: spec/support/models.rb [hunk 6]
```diff
@@ -21,6 +22,12 @@ class Teacher < ActiveRecord::Base
 class Student < ActiveRecord::Base
   has_many :classrooms
   has_many :teachers, through: :classrooms
+  
+  has_many :comments, as: :author
+  has_many :posts, through: :comments
+  
+  define_eager_group :posts_count, :posts, :count, "distinct post_id"
+
```
Comment (flyerhzm) [inline spec/support/models.rb]: no extra line break

**Comment by flyerhzm** [comment]
@yakjuly all look good except some minor issues, please fix them, then I'm happy to merge it.

**Comment by yakjuly** [comment]
@flyerhzm , beak lines have been removed.

**Comment by flyerhzm** [comment]
@yakjuly thanks for PR, I have released version 0.3.0 for this.

**Pull request merged**